tests/draft2020-12/dynamicRef.json

```diff
@@ -392,45 +392,61 @@
         "schema": {
             "$schema": "https://json-schema.org/draft/2020-12/schema",
             "$id": "https://test.json-schema.org/dynamic-ref-with-multiple-paths/main",
-            "$defs": {
-                "inner": {
-                    "$id": "inner",
-                    "$dynamicAnchor": "foo",
-                    "title": "inner",
-                    "additionalProperties": {
-                        "$dynamicRef": "#foo"
-                    }
-                }
-            },
             "if": {
                 "propertyNames": {
                     "pattern": "^[a-m]"
                 }
             },
             "then": {
-                "title": "any type of node",
-                "$id": "anyLeafNode",
-                "$dynamicAnchor": "foo",
+                "$id": "numberValues",
+                "$defs": {
+                  "foo": {
+                    "$dynamicAnchor": "foo",
+                    "type": "number"
+                  }
+                },
                 "$ref": "inner"
             },
             "else": {
-                "title": "integer node",
-                "$id": "integerNode",
-                "$dynamicAnchor": "foo",
-                "type": [ "object", "integer" ],
+                "$id": "stringValues",
+                "$defs": {
+                  "foo": {
+                    "$dynamicAnchor": "foo",
+                    "type": "string"
+                  }
+                },
                 "$ref": "inner"
+            },
+            "$defs": {
+                "inner": {
+                    "$id": "inner",
+                    "$dynamicAnchor": "foo",
+                    "additionalProperties": {
+                        "$dynamicRef": "#foo"
+                    }
+                }
             }
         },
         "tests": [
             {
-                "description": "recurse to anyLeafNode - floats are allowed",
+                "description": "alpha with number values",
```

Comment:
This test is/was indeed super difficult to figure out when implementing support, it's way more complicated than the average test, so full support on simplifying it -- but I actually remember finding the *data* unhelpful, because "alpha" and "november" come out of nowhere in the data and you have to kind of understand why there's a difference between them from a regex (albeit a simple one).

I think it'd probably also be even easier to follow if the `if` condition in the schema matched property names that were literally `shouldBeANumber` and `shouldBeAString` rather than being an unusual condition about what letter the word starts with.

Reply:
Agreed. I stopped after fixing the part that was incomprehensible, but it can definitely be simplified further.